
This patch reverts:

commit 3caaeffbe8ded4deaeb946d23631b04507bee408
Author: Jay Satiro <raysatiro@yahoo.com>
Date:   Fri May 20 16:44:01 2016 -0400

    openssl: cleanup must free compression methods
    
    - Free compression methods if OpenSSL 1.0.2 to avoid a memory leak.
    
    Bug: https://github.com/curl/curl/issues/817
    Reported-by: jveazey@users.noreply.github.com

diff --git b/lib/vtls/openssl.c a/lib/vtls/openssl.c
index 3a4bde5..6e3335e 100644
--- b/lib/vtls/openssl.c
+++ a/lib/vtls/openssl.c
@@ -747,11 +747,6 @@ void Curl_ossl_cleanup(void)
 
   /* Free all memory allocated by all configuration modules */
   CONF_modules_free();
-
-#if OPENSSL_VERSION_NUMBER >= 0x10002003L && \
-    OPENSSL_VERSION_NUMBER <= 0x10002FFFL
-  SSL_COMP_free_compression_methods();
-#endif
 }
 
 /*
